apps/e2e-tests/src/test-scenarios/wit/environment-setup.nix

```diff
@@ -0,0 +1,152 @@
+{
+  config,
+  lib,
+  ...
+}:
+let
+  # Function to read and parse the JSON file
+  readJson = path: builtins.fromJSON (builtins.readFile path);
+
+  readPortsFromFile =
+    path:
+    let
+      content = builtins.readFile path;
+      lines = lib.strings.splitString "\n" content;
+      nonEmpty = builtins.filter (s: s != "") lines;
+      asInts = builtins.map builtins.fromJSON nonEmpty;
+    in
+    asInts;
+
+  root = ../../../../..;
+
+  availablePorts =
+    let
+      filePath = "${config.devenv.root}/config/generated/process-compose/e2e-wit/available-ports";
+      ports = if builtins.pathExists filePath then readPortsFromFile filePath else [ ];
+    in
+    if builtins.length ports > 0 then ports else [ 8547 ];
+  testKeysDir = lib.path.append root "nix/test-environments/test-keys";
+  deploymentV2FilePath = lib.path.append root "config/evm_contracts_deployment_v2/ink-sepolia.json";
+
+  upgradeableProxyADFSContractAddressInk =
+    (readJson deploymentV2FilePath).contracts.coreContracts.UpgradeableProxyADFS.address;
+  impersonationAddress = lib.strings.fileContents "${testKeysDir}/impersonation_address";
+  anvilInkSepoliaPort = builtins.elemAt availablePorts 0;
+
+  apiKeysDir = "${config.devenv.root}/apps/e2e-tests/src/test-scenarios/wit/test-keys";
+in
+{
```

Comment:
I would import `../e2e-general/environment-setup.nix` here and override the settings that are different.